docs/profiling.md

```diff
@@ -0,0 +1,31 @@
+# vHive profiling tool
+
+## Methodology
+
+The benchmark loads a certain amount of requests per second (RPS) to vHive for a time duration.
+
+The load procedure is split into different steps from 5% to 100% of input RPS. If the tail latency at a load step violates the constraint which is 5x image service time, the tool will return the profiling result from one step before the tail latency violation.
+
+A load step divides into three parts: warm-up, profiling and cool-down. During the profile period, the benchmark records the average execution time of invocations and how many invocations return successfully. It also invokes [pmu-tools](https://github.com/andikleen/pmu-tools) to monitor user-defined nodes followed by [TopDown method](https://ieeexplore.ieee.org/document/6844459). After the measurement finishes, successful RPS per core, average execution time and the average counters of nodes will save in the `profile.csv`.
+
```

Comment:
you need to explain all important runtime arguments here so that a user of the tool is able to fine-tune them for their functions, server, sockets, cores, etc.

Reply:
Done